### PR TITLE
Test for deadlock

### PR DESCRIPTION
**Pull request by Stolb27**
According to https://github.com/greenplum-db/gpdb/pull/9985#issuecomment-618426312

### CODE REVIEW

File: src/test/regress/expected/fts_deadlock.out [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    while iterations < 120 loop /*in func*/
+        perform pg_sleep(1); /*in func*/
+        if exists (select * from gp_segment_configuration where content = 0 and mode = target_mode) then /*in func*/
+                return; /*in func*/
+        end if; /*in func*/
+        iterations := iterations + 1; /*in func*/
+    end loop; /*in func*/
+end $$ /*in func*/
+language plpgsql;
+-- Stop content 0 primary and let the mirror take over
```
Comment (Stolb27) [inline src/test/regress/expected/fts_deadlock.out]: Fix copypaste comment

File: src/test/regress/sql/fts_deadlock.sql [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+end $$ /*in func*/
+language plpgsql;
+
+-- Stop content 0 primary and let the mirror take over
+select stop_segment(fselocation) from pg_filespace_entry fe, gp_segment_configuration c, pg_filespace f
+where fe.fsedbid = c.dbid and c.content=0 and c.role='m' and f.oid = fe.fsefsoid and f.fsname = 'pg_system';
+
+select wait_for_content0('c');
+
+SELECT gp_inject_fault('add_segment_persistent_entries', 'sleep', '', '', '', 1, 70, 2::smallint);
```
Comment (darthunix) [inline src/test/regress/sql/fts_deadlock.sql]: Can you make `dbid` = `2` more obvious? Like `select dbid from gp_segment_configuration where content = 0 and mode = 'p';`?

File: src/test/regress/sql/fts_deadlock.sql [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+-- start_ignore
+\! gprecoverseg -a -F;
+-- end_ignore
+
+select wait_for_content0('s');
+
+select gp_inject_fault('add_segment_persistent_entries', 'reset', 2);
```
Comment (darthunix) [inline src/test/regress/sql/fts_deadlock.sql]: Same thing about `dbid 2` as in a comment above

File: src/test/regress/expected/fts_deadlock.out (hunk 1 above)
Comment (Stolb27) [reply src/test/regress/expected/fts_deadlock.out]: fixed

File: src/test/regress/sql/fts_deadlock.sql (hunk 2 above)
Comment (Stolb27) [reply src/test/regress/sql/fts_deadlock.sql]: Thanks!

File: src/test/regress/sql/fts_deadlock.sql (hunk 3 above)
Comment (Stolb27) [reply src/test/regress/sql/fts_deadlock.sql]: Fixed.

**Review by darthunix — APPROVED** [review]

**Comment by Stolb27** [comment]
Request for test cancelled.

**Pull request closed without merging**